lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -106,11 +110,31 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
                 return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
             case FIXED_LEN_BYTE_ARRAY:
                 return createDecimalColumnReader(descriptor)
-                        .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType()));
+                        .orElseGet(() -> {
+                            // Iceberg 0.11.1 writes UUID as FIXED_LEN_BYTE_ARRAY without logical type annotation (see https://github.com/apache/iceberg/pull/2913)
+                            if (isLogicalUuid(descriptor.getPrimitiveType()) || descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null) {
```

Comment:
Doing this annotation check inside the `orElseGet()` technically works, since `createDecimalColumnReader()` does a type check, but it seems better to do the annotation check first.

Reply:
that would lead to code duplication, since decimal case needs to be handled before `descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null` is checked.
unless we depend on https://github.com/apache/iceberg/pull/2913 and reject data sets created before that change.